test/function.localstack.test.ts

```diff
@@ -0,0 +1,332 @@
+import { clientConfig, localstackTestSuite } from "./localstack";
+import { $AWS, EventBus, EventBusRuleInput, Function } from "../src";
+import { Lambda } from "aws-sdk";
+import { aws_events, Stack } from "aws-cdk-lib";
+
+const lambda = new Lambda(clientConfig);
+
+localstackTestSuite("functionStack", (testResource, _stack, _app) => {
+  testResource(
+    "Call Lambda",
+    (parent) => {
+      const func = new Function(parent, "func2", async (event) => event);
+
+      return {
+        outputs: {
+          functionName: func.resource.functionName,
+        },
+      };
+    },
+    async (context) => {
+      await testFunction(context.functionName, {}, {});
+    }
+  );
+
+  testResource(
+    "Call Lambda from closure",
+    (parent) => {
+      const create = () =>
+        new Function(parent, "function", async (event) => event);
+
+      return { outputs: { function: create().resource.functionName } };
+    },
+    async (context) => {
+      await testFunction(context.function, {}, {});
+    }
+  );
+
+  testResource(
+    "Call Lambda from closure with variables",
+    (parent) => {
+      const create = () => {
+        const val = "a";
+        return new Function(parent, "function", async () => val);
+      };
+
+      return { outputs: { function: create().resource.functionName } };
+    },
+    async (context) => {
+      await testFunction(context.function, {}, "a");
```

Comment:
This seems boiler-platey and confusing. Can we simplify somehow?

```ts
const func = new Function(.., (param1, param2) => ..);

expect (await synth(func)(param1, param2)).toEqual(result);
```

Reply:
Was trying to reduce the number of stack deployments, it also supports automatic creation of outputs and passes them back to the test function. Deployments to `localstack` are not fast and would require either a unique state, or for the same stack to incrementally deploy during each test. 

The `outputs: { function: ... }` part is boilerplaty, but this wrapper supports any resource (there is example that uses the bus arns below). 

I'll see if I can make a special wrapper for the lambda tests to make them smaller.

What does `synth` do here?

Reply:
refactored

```ts
testFunctionResource("name", () => new Function(...), expected, payload?);
```